build/lib/path-helpers.js

```diff
@@ -8,12 +8,14 @@ const path = require('path');
 const pkgpath = require('packpath');
 
 const { directories, baseUrl: packageBaseUrl } = require(path.resolve(pkgpath.self(), 'package.json')); // eslint-disable-line
-
-
-const baseUrl = process.env.BASE_URL || packageBaseUrl || '/';
 const source = (...paths) => path.resolve(pkgpath.self(), directories.source, ...paths);
 const dest = (...paths) => path.resolve(pkgpath.self(), directories.dest, ...paths);
 
+// dev always serves from root (localhost:8080),
+// so only use baseUrl if task is not dev-related
+const devTasks = ['dev', 'build:dev', 'test:dev'];
+const baseUrl = devTasks.includes(process.env.npm_lifecycle_event) ? '/' : process.env.BASE_URL || packageBaseUrl || '/';
```

Comment:
Ternaries, especially those with a number of || statements in the else statement, are not the most legible code style. I'd like the see this written as a function instead. Or at the very least as an actual if statement. Basically I'd prefer anything that's easier for a dev to parse when scanning through the file when the build blows up.

Reply:
I've gone ahead and created a `getBaseUrl` function, as well as tidied up the comments so the file is more readable in general. Let me know if anything looks off.